compiler/src/dotty/tools/dotc/core/quoted/PickledQuotes.scala

```diff
@@ -52,7 +52,8 @@ object PickledQuotes {
   private def dealiasTypeTags(tp: Type)(implicit ctx: Context): Type = new TypeMap() {
     override def apply(tp: Type): Type = {
       val tp1 = tp match {
-        case tp: TypeRef if tp.typeSymbol.hasAnnotation(defn.InternalQuoted_QuoteTypeTagAnnot) => tp.dealias
+        case tp: TypeRef if tp.typeSymbol.hasAnnotation(defn.InternalQuoted_QuoteTypeTagAnnot) =>
+          tp.symbol.info.hiBound
```

Comment:
why this change?

Reply:
For some reason the type aliases get transformed into type bounds with equivalent upper and lower bound but they do not get dealiased. I will investigate further.